### PR TITLE
add pulpcore pipeline

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

File: pipelines/pulpcore/02-install.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+- name: install pulpcore
+  hosts:
+   - "{{ forklift_server_name }}"
+  become: True
+  vars_files:
+    - ../vars/install_base.yml
+    - ../vars/forklift_{{ pipeline_type }}.yml
+  roles:
+    - epel_repositories
+    - pulp_all_services
```
Comment (ehelms) [inline pipelines/pulpcore/02-install.yaml:11]: I think this role be missing.

Comment (evgeni) [reply pipelines/pulpcore/02-install.yaml:11]: no, it just needs the roles from pulp_installer and I didn't document that part yet

Comment (evgeni) [reply pipelines/pulpcore/02-install.yaml]: now it installs it automatically

**Review by mikedep333 — APPROVED** [review]

File: playbooks/setup_forklift.yml [hunk 2]
```diff
@@ -41,3 +41,13 @@
...
+
+    - name: 'Install Forklift collection dependencies'
+      command:
+        cmd: ansible-galaxy collection install -r requirements.yml
+        chdir: "{{ forklift_dest }}"
+
+    - name: 'Install Forklift pulp_installer role dependencies'
+      command:
+        cmd: ansible-galaxy role install -r playbooks/galaxy_collections/ansible_collections/pulp/pulp_installer/requirements.yml
+        chdir: "{{ forklift_dest }}"
```
Comment (ehelms) [inline playbooks/setup_forklift.yml:53]: Ought this be done every time? Perhaps a variable would be the way to go here so th epipeline can set it?

Comment (evgeni) [reply playbooks/setup_forklift.yml:53]: you mean putting this in a block/role and add a `when`? yeah, sure, why not.

File: playbooks/setup_forklift.yml [hunk 3]
```diff
@@ -41,3 +42,15 @@
...
+      command:
+        cmd: ansible-galaxy collection install -r requirements.yml
+        chdir: "{{ forklift_dest }}"
+      when: forklift_install_from_galaxy
+
+    - name: 'Install Forklift pulp_installer role dependencies'
+      command:
+        cmd: ansible-galaxy role install -r playbooks/galaxy_collections/ansible_collections/pulp/pulp_installer/requirements.yml
+        chdir: "{{ forklift_dest }}"
+      when: forklift_install_from_galaxy
```
Comment (ehelms) [inline playbooks/setup_forklift.yml:56]: We can change this later, but I forsee a future where we will want a specific install from galaxy option for just Pulp assets.

Comment (evgeni) [reply playbooks/setup_forklift.yml:56]: I'd prefer to change that when required.

**Comment by evgeni** [comment]
@ehelms was there anything else you wanted addressed here?

File: pipelines/vars/forklift_pulpcore.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
+server_box:
+  box: "{{ pipeline_os }}"
+  memory: 4680
+
+forklift_boxes:
+  "{{ {forklift_server_name: server_box} }}"
+
+pulp_install_source: packages
+pulp_pkg_repo: "https://yum.theforeman.org/pulpcore/{{ pipeline_version }}/el{{ ansible_distribution_major_version }}/$basearch/"
+# staging: http://koji.katello.org/releases/yum/pulpcore-{{ pipeline_version }}/el{{ ansible_distribution_major_version }}/$basearch/
```
Comment (ehelms) [inline pipelines/vars/forklift_pulpcore.yml:10]: Intentional to leave this commented out here?

File: pipelines/pulpcore/02-install.yaml [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+  hosts:
+   - "{{ forklift_server_name }}"
+  become: True
+  vars_files:
+    - ../vars/install_base.yml
+    - ../vars/forklift_{{ pipeline_type }}.yml
+  roles:
+    - epel_repositories
+    - pulp.pulp_installer.pulp_all_services
+  environment:
```
Comment (ehelms) [inline pipelines/pulpcore/02-install.yaml:12]: Tis a nit, but I'd tend to expect seeing this above roles.

File: pipelines/vars/forklift_pulpcore.yml (hunk 4 above)
Comment (evgeni) [reply pipelines/vars/forklift_pulpcore.yml:10]: No, actually, for the pipeline, it should have been the staging repo to begin with ;)

File: pipelines/pulpcore/02-install.yaml (hunk 5 above)
Comment (evgeni) [reply pipelines/pulpcore/02-install.yaml:12]: fair, will update

**Review by ehelms — APPROVED** [review]

**Pull request merged**